drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -386,149 +281,170 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+		LL_RCC_MSI_EnableRangeSelection();
 
-	/* Set MSIS as SYSCLCK source */
-	set_up_clk_msis();
-	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_MSIS);
-	LL_RCC_SetAHBPrescaler(s_ClkInitStruct.AHBCLKDivider);
-	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_MSIS) {
-	}
+		LL_RCC_MSIK_SetRange(STM32_MSIK_RANGE << RCC_ICSCR1_MSIKRANGE_Pos);
 
-	/* Update SystemCoreClock variable */
-	LL_SetSystemCoreClock(new_hclk_freq);
+		if (IS_ENABLED(STM32_MSIK_PLL_MODE)) {
+			__ASSERT(STM32_LSE_ENABLED,
+				"MSIK Hardware auto calibration needs LSE clock activation");
+			/* Enable MSI hardware auto calibration */
+			LL_RCC_MSI_EnablePLLMode();
+		}
 
-	/* Set peripheral busses prescalers */
-	LL_RCC_SetAPB1Prescaler(s_ClkInitStruct.APB1CLKDivider);
-	LL_RCC_SetAPB2Prescaler(s_ClkInitStruct.APB2CLKDivider);
-	LL_RCC_SetAPB3Prescaler(s_ClkInitStruct.APB3CLKDivider);
+		if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+			__ASSERT((STM32_MSIK_PLL_MODE == STM32_MSIS_PLL_MODE),
+				"Please check MSIS/MSIK config consistency");
+		}
 
-	/* If freq not increased, set flash latency after all clock setting */
-	if (new_hclk_freq <= old_hclk_freq) {
-		LL_SetFlashLatency(new_hclk_freq);
+		/* Enable MSIK */
+		LL_RCC_MSIK_Enable();
+
+		/* Wait till MSIK is ready */
+		while (LL_RCC_MSIK_IsReady() != 1) {
+		}
 	}
 
-	/* Disable other clocks */
-	LL_RCC_HSE_Disable();
-	LL_RCC_HSI_Disable();
-	LL_RCC_PLL1_Disable();
+	if (IS_ENABLED(STM32_LSI_ENABLED)) {
+		/* Enable LSI oscillator */
+		LL_RCC_LSI_Enable();
+		while (LL_RCC_LSI_IsReady() != 1) {
+		}
+	}
 
-#endif	/* STM32_SYSCLK_SRC_MSIS */
 }
 
-/*
- * Configure HSI as source of SYSCLK
- */
-void config_src_sysclk_hsi(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+int stm32_clock_control_init(const struct device *dev)
 {
-#ifdef STM32_SYSCLK_SRC_HSI
-
-	clock_switch_to_hsi(s_ClkInitStruct.AHBCLKDivider);
-
-	/* Update SystemCoreClock variable */
-	LL_SetSystemCoreClock(__LL_RCC_CALC_HCLK_FREQ(HSI_VALUE,
-						s_ClkInitStruct.AHBCLKDivider));
+	uint32_t old_hclk_freq = 0;
+	uint32_t new_hclk_freq = 0;
+	int r = 0;
 
-	/* Set peripheral busses prescalers */
-	LL_RCC_SetAPB1Prescaler(s_ClkInitStruct.APB1CLKDivider);
-	LL_RCC_SetAPB2Prescaler(s_ClkInitStruct.APB2CLKDivider);
-	LL_RCC_SetAPB3Prescaler(s_ClkInitStruct.APB3CLKDivider);
+	ARG_UNUSED(dev);
 
-	/* Set flash latency */
-	/* HSI used as SYSCLK, set latency to 0 */
-	LL_FLASH_SetLatency(LL_FLASH_LATENCY_0);
+	/* Set up indiviual enabled clocks */
+	set_up_fixed_clock_sources();
 
-	/* Disable other clocks */
-	LL_RCC_HSE_Disable();
-	LL_RCC_MSIS_Disable();
-	LL_RCC_PLL1_Disable();
-
-#endif	/* STM32_SYSCLK_SRC_HSI */
-}
+	/* Set up PLLs */
+	r = set_up_plls();
+	if (r < 0) {
+		return r;
+	}
 
-int stm32_clock_control_init(const struct device *dev)
-{
-	LL_UTILS_ClkInitTypeDef s_ClkInitStruct;
+	/* Set voltage regulator to comply with targeted system frequency */
+	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
 
-	ARG_UNUSED(dev);
```

Comment:
still applicable

Reply:
It's not removed, but moved few lines above.
(well, actually not, but this is how github display changes when adding squashing all the changes, this should be clearer in individual commits)